elytron/src/test/java/org/wildfly/extension/elytron/LdapTestCase.java

```diff
@@ -62,13 +79,72 @@
+        KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+        KeyPair scarabKeys = keyPairGenerator.generateKeyPair();
+        PublicKey scarabPublicKey = scarabKeys.getPublic();
+
+        SelfSignedX509CertificateAndSigningKey issuerSelfSignedX509CertificateAndSigningKey = SelfSignedX509CertificateAndSigningKey.builder()
+                .setDn(ISSUER_DN)
+                .setKeyAlgorithmName("RSA")
+                .setSignatureAlgorithmName("SHA1withRSA")
+                .addExtension(false, "BasicConstraints", "CA:true,pathlen:2147483647")
+                .build();
+
+        return new X509CertificateBuilder()
+                .setIssuerDn(ISSUER_DN)
+                .setSubjectDn(SCARAB_DN)
+                .setSignatureAlgorithmName("SHA1withRSA")
+                .setSigningKey(issuerSelfSignedX509CertificateAndSigningKey.getSigningKey())
+                .setPublicKey(scarabPublicKey)
+                .setSerialNumber(new BigInteger("4"))
+                .addExtension(new BasicConstraintsExtension(false, false, -1))
+                .build();
+    }
+
+    private static void rewriteLdapDataLdif(X509Certificate certificate) throws Exception {
+        MessageDigest md = MessageDigest.getInstance("SHA-1");
+        String digest = ByteIterator.ofBytes(md.digest(certificate.getEncoded())).hexEncode(true).drainToString();
+
+        CodePointIterator certificateBytes = ByteIterator.ofBytes(certificate.getEncoded()).base64Encode();
+        String certificateBaseString = "usercertificate:: " + certificateBytes.drainToString();
+        String certificateString = "";
+        int counter = 0;
+        for (int i = 0; i < certificateBaseString.length(); i++){
+            if(i == 78 || i == (78+77*counter)){
+                certificateString = certificateString + System.getProperty("line.separator");
+                certificateString = certificateString + " ";
+                counter += 1;
+            }
+            certificateString = certificateString + certificateBaseString.charAt(i);
+        }
+
+        // Read in the file and then replace the x509digest and usercertificate
+        Path filePath = Paths.get(WORKING_DIRECTORY_LOCATION + LDIF_LOCATION);
+        String ldapDataLdifString = new String(Files.readAllBytes(filePath),StandardCharsets.UTF_8);
+        ldapDataLdifString = ldapDataLdifString.replaceFirst("x509digest: (.*)", "x509digest: " + digest);
+        ldapDataLdifString = ldapDataLdifString.replaceFirst(Pattern.quote("usercertificate::"), certificateString);
+        Files.write(filePath, ldapDataLdifString.getBytes(StandardCharsets.UTF_8));
+    }
+
+    private static void loadResources() throws Exception {
+        SCARAB_CERTIFICATE = createScarabCertificate();
+        rewriteLdapDataLdif(SCARAB_CERTIFICATE);
+    }
 
     @BeforeClass
-    public static void startLdapService() {
+    public static void startLdapService() throws Exception {
+        loadResources();
```

Comment:
Looks like this test class is missing code to delete the test files created by ```TestEnvironment.startLdapService()```.

Reply:
Fixed